android/src/main/java/io/getconnect/client/android/AndroidConnectClient.java

```diff
@@ -17,4 +20,14 @@ public AndroidConnectClient(Context context, String projectId, String apiKey) th
     private AndroidConnectClient(Context context, String projectId, String apiKey, JsonSerializer serializer) throws IOException {
         super(projectId, apiKey, null, new UrlConnectionHttpClient(), serializer, new FileEventStore(projectId, context.getCacheDir(), serializer));
     }
+
+    /**
+     * Encrypt filtered key for use with the Connect API.
+     * @param key the definition of the filtered key.
+     * @param masterKey the master key for the Connect project
+     * @return The encrypted filtered key.
+     */
+    public static String generateFilteredKey(final Map<String, Object> key, String masterKey) throws FilteredKeyException {
+        return FilteredKey.encrypt(new AndroidJsonSerializer(), key, masterKey);

```

Comment:
We should use the serializer that is stored from the constructor so it can be overriden as needed.